File: src/main/java/dev/sky_lock/pocketlifevehicle/gui/contents/ItemSelectContents.java
package dev.sky_lock.pocketlifevehicle.gui.contents;

import dev.sky_lock.menu.InventoryMenu;
import dev.sky_lock.menu.MenuContents;
import dev.sky_lock.menu.Slot;
import dev.sky_lock.pocketlifevehicle.PLVehicle;
import dev.sky_lock.pocketlifevehicle.gui.EditSessions;
import dev.sky_lock.pocketlifevehicle.gui.ModelMenuIndex;
import dev.sky_lock.pocketlifevehicle.item.ItemStackBuilder;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;

/**
 * @author sky_lock
 */

public class ItemSelectContents extends MenuContents {

    public ItemSelectContents(Player player) {
        short idOffset = 1000;
        short k = 1;
        for (int i = 0; i < 6; i++) {
            for (int j = 0; j < 9; j++) {
                int id = idOffset + k;
                ItemStack item = ItemStackBuilder.of(PLVehicle.CAR_ITEM, 1).customModelData(id).build();
                this.addSlot(new Slot(i * 9 + j, item, (event) -> {
                    EditSessions.of(player.getUniqueId()).ifPresent(session -> {
                        session.setItemType(PLVehicle.CAR_ITEM);
                        session.setItemID(id);
                    });
                    flipPage(player, ModelMenuIndex.ITEM_OPTION.ordinal());
                }));
                k++;
            }
        }
    }

    @Override
    public void onFlip(InventoryMenu inventoryMenu) {

    }

    private void flipPage(Player player, int page) {
        InventoryMenu.of(player).ifPresent(menu -> {
            menu.flip(player, page);
        });
    }

}

File: src/main/kotlin/dev/sky_lock/pocketlifevehicle/gui/contents/ItemSelectContents.kt
package dev.sky_lock.pocketlifevehicle.gui.contents

import dev.sky_lock.menu.InventoryMenu
import dev.sky_lock.menu.InventoryMenu.Companion.of
import dev.sky_lock.menu.MenuContents
import dev.sky_lock.menu.Slot
import dev.sky_lock.pocketlifevehicle.gui.EditSessions.of
import dev.sky_lock.pocketlifevehicle.gui.ModelMenuIndex
import dev.sky_lock.pocketlifevehicle.gui.ModelOption
import dev.sky_lock.pocketlifevehicle.item.ItemStackBuilder
import org.bukkit.Material
import org.bukkit.entity.Player

/**
 * @author sky_lock
 */
class ItemSelectContents(player: Player) : MenuContents() {
    private val VEHICLE_ITEM = Material.IRON_NUGGET
    override fun onFlip(menu: InventoryMenu) {}
    private fun flipPage(player: Player, page: Int) {
        of(player).ifPresent { menu: InventoryMenu -> menu.flip(player, page) }
    }

    init {
        val idOffset: Short = 1000
        var k: Short = 1
        for (i in 0..5) {
            for (j in 0..8) {
                val id = idOffset + k
                val item = ItemStackBuilder(VEHICLE_ITEM, 1).customModelData(id).build()
                addSlot(Slot(i * 9 + j, item, org.bukkit.util.Consumer {
                    of(player.uniqueId).ifPresent { session: ModelOption ->
                        session.itemType = VEHICLE_ITEM
                        session.setItemID(id)
                    }
                    flipPage(player, ModelMenuIndex.ITEM_OPTION.ordinal)
                }))
                k++
            }
        }
    }
}